packages/react-components/react-table/stories/Table/Virtualized.stories.tsx

```diff
@@ -0,0 +1,236 @@
+    lastUpdate: {
+      label: 'You edited this',
+      icon: <EditRegular />,
+    },
+  },
+  {
+    file: { label: 'Thursday presentation', icon: <FolderRegular /> },
+    author: { label: 'Erika Mustermann', status: 'busy' },
+    lastUpdated: { label: 'Yesterday at 1:45 PM', timestamp: 2 },
+    lastUpdate: {
+      label: 'You recently opened this',
+      icon: <OpenRegular />,
+    },
+  },
+  {
+    file: { label: 'Training recording', icon: <VideoRegular /> },
+    author: { label: 'John Doe', status: 'away' },
+    lastUpdated: { label: 'Yesterday at 1:45 PM', timestamp: 2 },
+    lastUpdate: {
+      label: 'You recently opened this',
+      icon: <OpenRegular />,
+    },
+  },
+  {
+    file: { label: 'Purchase order', icon: <DocumentPdfRegular /> },
+    author: { label: 'Jane Doe', status: 'offline' },
+    lastUpdated: { label: 'Tue at 9:30 AM', timestamp: 1 },
+    lastUpdate: {
+      label: 'You shared this in a Teams chat',
+      icon: <PeopleRegular />,
+    },
+  },
+];
+
+const repeatCount = 2500;
+const generateContent = (): Item[] => {
+  const contentList: Item[] = [];
+  for (let i = 0; i < repeatCount; i++) {
+    items.forEach((item, index) => {
+      const newItem: Item = {
+        file: { ...item.file },
+        author: { ...item.author },
+        lastUpdated: { ...item.lastUpdated },
+        lastUpdate: { ...item.lastUpdate },
+      };
+      // Update labels to make unique
+      newItem.author.label = `${newItem.author.label}-${i}-${index}`;
+      newItem.file.label = `${newItem.file.label}-${i}-${index}`;
+      newItem.lastUpdate.label = `${newItem.lastUpdate.label}-${i}-${index}`;
+      newItem.lastUpdated.label = `${newItem.lastUpdated.label}-${i}-${index}`;
+      contentList.push(newItem);
+    });
+  }
+  return contentList;
+};
+
+const fullItemList: Item[] = generateContent();
+
+// Simple wrapper class to control scroll view height
+const useStyles = makeStyles({
+  container: {
+    display: 'flex',
+    flexDirection: 'column',
+    overflowAnchor: 'none',
+    overflowY: 'auto',
+    width: '100%',
+    height: '100%',
+    maxHeight: '750px',
+  },
+});
+
+export const Virtualized = () => {
+  const columns: ColumnDefinition<Item>[] = React.useMemo(
+    () => [
+      createColumn<Item>({
+        columnId: 'file',
+        compare: (a, b) => {
+          return a.file.label.localeCompare(b.file.label);
+        },
+      }),
+      createColumn<Item>({
+        columnId: 'author',
+        compare: (a, b) => {
+          return a.author.label.localeCompare(b.author.label);
+        },
+      }),
+      createColumn<Item>({
+        columnId: 'lastUpdated',
+        compare: (a, b) => {
+          return a.lastUpdated.timestamp - b.lastUpdated.timestamp;
+        },
+      }),
+      createColumn<Item>({
+        columnId: 'lastUpdate',
+        compare: (a, b) => {
+          return a.lastUpdate.label.localeCompare(b.lastUpdate.label);
+        },
+      }),
+    ],
+    [],
+  );
```

Comment:
Per #26374 can you hoist the `columns` outside of the component and remove `useMemo`?

Reply:
Moving this to a separate PR for table demo